node/src/main/scala/com/wavesplatform/api/http/eth/EthRpcRoute.scala

```diff
@@ -69,131 +69,160 @@ class EthRpcRoute(blockchain: Blockchain, transactionsApi: CommonTransactionsApi
         case "eth_getTransactionCount" =>
           resp(id, quantity(time.getTimestamp()))
         case "eth_getBlockByNumber" =>
-          resp(
-            id,
-            Json.obj(
-              "number" -> quantity(Integer.parseInt(param1Str.drop(2), 16))
-            )
-          )
+          param1StrE
+            .map { str =>
+              val blockNumber = str match {
+                case "earliest" => Some(1)
+                case "latest"   => Some(blockchain.height)
+                case "pending"  => None
+                case _          => Some(Integer.parseInt(str.drop(2), 16))
```

Comment:
We validate params above, so it would be consistent to validate a parsed number here too (e.g. `Try(...)`), right?